app/assets/javascripts/directives/file-upload-modal.coffee

```diff
@@ -0,0 +1,56 @@
+angular.module('infish').directive 'fileUploadModal', ($timeout) ->
+  templateUrl: 'file-upload-modal.html'
+  controller: ($scope) ->
+    modalOpen = false
+
+    $scope.ifShowModal = -> modalOpen
+    $scope.showModal = -> modalOpen = true
+    $scope.closeModal = -> modalOpen = false
+    $scope.toggleModal = -> modalOpen = !modalOpen
+
+
+    addToForm = (question, answers) ->
+      states = []
+      states.push (question[i] == '1') for i in [1..4]
+      question = question[5]
```

Comment:
Output from regex. First four matches are correction indicators, fifth is a question.